pygfx/renderers/wgpu/meshrender.py

```diff
@@ -488,6 +491,29 @@ def helpers(self):
             return albeido * (ambient_color + diffuse_color) + specular_color;
         }
 
+        fn lighting_flat(
+            is_front: bool,
+            world_pos: vec3<f32>,
+            normal: vec3<f32>,
+            light: vec3<f32>,
+            view: vec3<f32>,
+            albeido: vec3<f32>,
+        ) -> vec3<f32> {
+
+            let u = dpdx(world_pos);
+            let v = dpdy(world_pos);
+            var normal = normalize(cross(u, v));
+
+            // The normal calculated above may not be oriented correctly.
+            // We have two flags: is_front and u_stdinfo.flipped_winding.
+            // Note that lighting_phong() also applies the is_front flag.
+            // Below code means: flip the normal if the XOR of these two flags is true.
+            normal = select(normal, -normal, (select(0, 1, is_front) + u_stdinfo.flipped_winding) == 1);
```

Comment:
I don't have a good way to explain this, but I tried this on the validate_culling.py example (using the flat material instead), and this is what I found makes it work :)